models/contacts.js

```diff
@@ -1,19 +1,80 @@
-// const fs = require('fs/promises')
+import {Schema, model} from "mongoose";
+import { handleSaveError, validateAtUpdate } from "./hooks.js";
+import ctrlWrapper from "../decorators/ctrlWrapper.js";
 
-const listContacts = async () => {}
+export const contactsSchema = new Schema({
+  name: {
+    type: String,
+    required: [true, 'Set name for contact'],
+  },
+  email: {
+    type: String,
+  },
+  phone: {
+    type: String,
+  },
+  favorite: {
+    type: Boolean,
+    default: false,
+  },
+}, {versionKey: false, timestamps: true})
 
-const getContactById = async (contactId) => {}
+contactsSchema.pre("findOneAndUpdate", validateAtUpdate);
+contactsSchema.post("save", handleSaveError);
+contactsSchema.post("findOneAndUpdate", handleSaveError);
 
-const removeContact = async (contactId) => {}
+const Contact = model("contact", contactsSchema);
 
-const addContact = async (body) => {}
+export const listContacts = async (req, res) => {
```

Comment:
this and next functions should be in controller, models its only about db models